src/NuGet.Core/NuGet.LibraryModel/LibraryIncludeFlagUtils.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace NuGet.LibraryModel
+{
+    /// <summary>
+    /// Helper methods for dealing withi include/exclude flag strings.
+    /// </summary>
+    public class LibraryIncludeFlagUtils
+    {
+        /// <summary>
+        /// By default build, contentFiles, and analyzers do not flow transitively between projects.
+        /// </summary>
+        public static LibraryIncludeFlags DefaultSuppressParent =
+            (LibraryIncludeFlags.Build | LibraryIncludeFlags.ContentFiles | LibraryIncludeFlags.Analyzers);
+
+        /// <summary>
+        /// Convert set of flag strings into a LibraryIncludeFlags.
+        /// </summary>
+        public static LibraryIncludeFlags GetFlags(IEnumerable<string> flags)
+        {
+            if (flags == null)
+            {
+                throw new ArgumentNullException(nameof(flags));
+            }
+
+            var result = LibraryIncludeFlags.None;
+
+            foreach (var flag in flags)
+            {
+                switch (flag.ToLowerInvariant())
```

Comment:
Should you use Enum.TryParse(string, out enumValue)? You would not need a switch statement then
You can simply result |= enumValue